From d7622bd55974947825a1c4b423db1895e38298e8 Mon Sep 17 00:00:00 2001
From: montellese <montellese@kodi.tv>
Date: Fri, 22 May 2020 23:52:41 +0200
Subject: [PATCH] fix focusing proper setting control in
 CGUIDialogContentSettings

Calls to SetFocus() were resolved to CGUIControl::SetFocus() instead of
CGUIDialogContentSettings::SetFocus().
---
 xbmc/settings/dialogs/GUIDialogContentSettings.cpp | 6 +++---
 xbmc/settings/dialogs/GUIDialogContentSettings.h   | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/settings/dialogs/GUIDialogContentSettings.cpp b/xbmc/settings/dialogs/GUIDialogContentSettings.cpp
index b66aa376f0cc..746678ced1e8 100644
--- a/xbmc/settings/dialogs/GUIDialogContentSettings.cpp
+++ b/xbmc/settings/dialogs/GUIDialogContentSettings.cpp
@@ -227,7 +227,7 @@ void CGUIDialogContentSettings::OnSettingAction(std::shared_ptr<const CSetting>
       m_scraper = std::dynamic_pointer_cast<CScraper>(scraperAddon);
 
       SetupView();
-      SetFocus(SETTING_CONTENT_TYPE);
+      SetFocusToSetting(SETTING_CONTENT_TYPE);
     }
   }
   else if (settingId == SETTING_SCRAPER_LIST)
@@ -246,7 +246,7 @@ void CGUIDialogContentSettings::OnSettingAction(std::shared_ptr<const CSetting>
       m_scraper = std::dynamic_pointer_cast<CScraper>(scraperAddon);
 
       SetupView();
-      SetFocus(SETTING_SCRAPER_LIST);
+      SetFocusToSetting(SETTING_SCRAPER_LIST);
     }
   }
   else if (settingId == SETTING_SCRAPER_SETTINGS)
@@ -399,7 +399,7 @@ void CGUIDialogContentSettings::ToggleState(const std::string &settingid, bool e
   }
 }
 
-void CGUIDialogContentSettings::SetFocus(const std::string &settingid)
+void CGUIDialogContentSettings::SetFocusToSetting(const std::string& settingid)
 {
   BaseSettingControlPtr settingControl = GetSettingControl(settingid);
   if (settingControl != NULL && settingControl->GetControl() != NULL)
diff --git a/xbmc/settings/dialogs/GUIDialogContentSettings.h b/xbmc/settings/dialogs/GUIDialogContentSettings.h
index 2ba57193b6f9..18770ac1b987 100644
--- a/xbmc/settings/dialogs/GUIDialogContentSettings.h
+++ b/xbmc/settings/dialogs/GUIDialogContentSettings.h
@@ -64,7 +64,7 @@ class CGUIDialogContentSettings : public CGUIDialogSettingsManualBase
   void SetLabel2(const std::string &settingid, const std::string &label);
   void ToggleState(const std::string &settingid, bool enabled);
   using CGUIDialogSettingsManualBase::SetFocus;
-  void SetFocus(const std::string &settingid);
+  void SetFocusToSetting(const std::string& settingid);
 
   /*!
   * @brief The currently selected content type
